modules/core/02-client/migrations/v7/store_test.go

```diff
@@ -0,0 +1,169 @@
+import (
+	"strconv"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/ibc-go/v6/modules/core/02-client/migrations/v7"
+	"github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	host "github.com/cosmos/ibc-go/v6/modules/core/24-host"
+	solomachine "github.com/cosmos/ibc-go/v6/modules/light-clients/06-solomachine"
+	ibctm "github.com/cosmos/ibc-go/v6/modules/light-clients/07-tendermint"
+	ibctesting "github.com/cosmos/ibc-go/v6/testing"
+)
+
+// numCreations is the number of clients/consensus states created for
+// solo machine and localhost clients
+const numCreations = 10
+
+type MigrationsV7TestSuite struct {
+	suite.Suite
+
+	coordinator *ibctesting.Coordinator
+
+	chainA *ibctesting.TestChain
+	chainB *ibctesting.TestChain
+}
+
+func (suite *MigrationsV7TestSuite) SetupTest() {
+	suite.coordinator = ibctesting.NewCoordinator(suite.T(), 2)
+
+	suite.chainA = suite.coordinator.GetChain(ibctesting.GetChainID(1))
+	suite.chainB = suite.coordinator.GetChain(ibctesting.GetChainID(2))
+}
+
+func TestIBCTestSuite(t *testing.T) {
+	suite.Run(t, new(MigrationsV7TestSuite))
+}
+
+// test that MigrateStore returns an error if the codec used doesn't register tendermint types
+func (suite *MigrationsV7TestSuite) TestMigrateStoreTendermint() {
+	path := ibctesting.NewPath(suite.chainA, suite.chainB)
+	suite.coordinator.SetupClients(path)
+
+	registry := codectypes.NewInterfaceRegistry()
+	cdc := codec.NewProtoCodec(registry)
+
+	solomachine.RegisterInterfaces(registry)
+	err := v7.MigrateStore(suite.chainA.GetContext(), suite.chainA.GetSimApp().GetKey(host.StoreKey), cdc)
+	suite.Require().Error(err)
+
+	ibctm.RegisterInterfaces(registry)
+	err = v7.MigrateStore(suite.chainA.GetContext(), suite.chainA.GetSimApp().GetKey(host.StoreKey), cdc)
+	suite.Require().NoError(err)
+}
+
+// create multiple solo machine clients, tendermint and localhost clients
+// ensure that solo machine clients are migrated and their consensus states are removed
+// ensure the localhost is deleted entirely.
+func (suite *MigrationsV7TestSuite) TestMigrateStore() {
+	paths := []*ibctesting.Path{
+		ibctesting.NewPath(suite.chainA, suite.chainB),
+		ibctesting.NewPath(suite.chainA, suite.chainB),
+	}
+
+	// create tendermint clients
+	for _, path := range paths {
+		suite.coordinator.SetupClients(path)
+	}
+
+	solomachines := []*ibctesting.Solomachine{
+		ibctesting.NewSolomachine(suite.T(), suite.chainA.Codec, "06-solomachine-0", "testing", 1),
+		ibctesting.NewSolomachine(suite.T(), suite.chainA.Codec, "06-solomachine-1", "testing", 4),
+	}
+
+	suite.createSolomachineClients(solomachines)
+	suite.createLocalhostClients()
+
+	err := v7.MigrateStore(suite.chainA.GetContext(), suite.chainA.GetSimApp().GetKey(host.StoreKey), suite.chainA.App.AppCodec())
+	suite.Require().NoError(err)
+
+	suite.assertSolomachineClients(solomachines)
+	suite.assertLocalhostClients()
```

Comment:
[nit] I think it would be nice to have what is being asserted clear in the testcase, i.e. `suite.assertNumLocalhostClients(0)`  or `suite.assertNoLocalhostClients()` rather than digging into the function to see what it is doing.

Reply:
agreed! Renamed to `assertNoLocalhostClients`